new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Ingress.java

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.weblogic.kubernetes.actions.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import io.kubernetes.client.custom.IntOrString;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1HTTPIngressPath;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1HTTPIngressRuleValue;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1Ingress;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1IngressBackend;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1IngressList;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1IngressRule;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1IngressSpec;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import oracle.weblogic.kubernetes.actions.impl.primitive.Kubernetes;
+
+import static oracle.weblogic.kubernetes.actions.ActionConstants.INGRESS_API_VERSION;
+import static oracle.weblogic.kubernetes.actions.ActionConstants.INGRESS_KIND;
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+
+/**
+ * Utility class for Voyager ingress controller.
+ */
+public class Ingress {
+
+  /**
+   * Create an ingress for the WebLogic domain with domainUid in the specified domain namespace.
+   * The ingress host is set to 'domainUid.clusterName.test'.
+   *
+   * @param ingressName name of the ingress to be created
+   * @param domainNamespace the WebLogic domain namespace in which the ingress will be created
+   * @param domainUid the WebLogic domainUid which is backend to the ingress
+   * @param clusterNameMsPortMap the map with key as cluster name and value as managed server port of the cluster
+   * @return list of ingress hosts or null if got ApiException when calling Kubernetes client API to create ingress
+   */
+  public static List<String> createIngress(String ingressName,
```

Comment:
missing javadoc for annotations argument

Reply:
fixed